x-pack/plugins/secops/server/lib/suricata/types.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SourceConfiguration, SuricataEvents, TimerangeInput } from '../../../common/graphql/types';
+import { JsonObject } from '../../../common/typed_json';
+import { FrameworkRequest } from '../framework';
+
+export interface SuricataAdapter {
+  getEvents(req: FrameworkRequest, options: SuricataRequestOptions): Promise<SuricataEvents[]>;
+}
+
+export type ESQuery = ESRangeQuery | ESQueryStringQuery | ESMatchQuery | JsonObject;
```

Comment:
Should these types be defined outside of the `suricata` types?

Reply:
That's the plan when we have more than one query.